app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -107,11 +92,38 @@ class FirebaseAuthRepository : AuthRepository {
     return intent
   }
 
-  /**
-   * Signs out the current user from Firebase and invokes the success callback.
-   *
-   * @param onSuccess Callback to invoke after the user has successfully signed out.
-   */
+  override fun createAccountWithEmailAndPassword(
+      onSuccess: (FirebaseUser?) -> Unit,
+      onErrorAction: (Exception) -> Unit,
+      email: String,
+      password: String
+  ) {
+    val auth = FirebaseAuth.getInstance()
+    auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        onSuccess(auth.currentUser)
+      } else {
+        onErrorAction(task.exception ?: Exception())
+      }
+    }
+  }
+
+  override fun signInWithEmailAndPassword(
+      onSuccess: (FirebaseUser?) -> Unit,
+      onErrorAction: (Exception) -> Unit,
+      email: String,
+      password: String
+  ) {
+    val auth = FirebaseAuth.getInstance()
+    auth.signInWithEmailAndPassword(email, password).addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        onSuccess(auth.currentUser)
+      } else {
+        onErrorAction(task.exception ?: Exception())
+      }
+    }
+  }
```

Comment:
Comment: Oh wow, this was indeed even more simple than I had imagined. Good job!